projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -130,29 +132,28 @@ StatusCode ads1015_init(Ads1015Storage *storage, I2CPort i2c_port, Ads1015Addres
 // This function enable/disables channels, and registers callbacks for each channel.
 StatusCode ads1015_configure_channel(Ads1015Storage *storage, Ads1015Channel channel, bool enable,
                                      Ads1015Callback callback, void *context) {
-  gpio_it_mask_interrupt(&storage->ready_pin, true);
   if (storage == NULL || channel >= NUM_ADS1015_CHANNELS) {
     return status_code(STATUS_CODE_INVALID_ARGS);
   }
+  gpio_it_mask_interrupt(&storage->ready_pin, true);
+
   uint8_t channel_enable_bitset = storage->channel_enable_bitset;
   prv_mark_channel_enabled(channel, enable, &storage->channel_enable_bitset);
   storage->channel_enable_bitset_pending = storage->channel_enable_bitset;
 
-  if (channel_enable_bitset == ADS1015_BITSET_EMPTY && enable) {
+  if ((channel_enable_bitset == ADS1015_BITSET_EMPTY) && enable) {
     // Activate the interrupt since the first channel is being enabled.
     status_ok_or_return(prv_set_channel(storage, channel));
     status_ok_or_return(gpio_it_mask_interrupt(&storage->ready_pin, false));
   } else if (!enable) {
     storage->channel_readings[channel] = ADS1015_DISABLED_CHANNEL_READING;
-   // printf("channel:%d disabled\n", channel);
   }
   gpio_it_mask_interrupt(&storage->ready_pin, false);
```

Comment:
I'm not sure if this is the right place to put this - couldn't you just combine this with line 147 and 153?